### PR TITLE
ASC-224 space request

### PR DESCRIPTION
**Pull request by mfrancisc**
Add new SpaceRequest CRD

See Jira ticket: https://issues.redhat.com/browse/ASC-244

Related PRs:

- member-operator: https://github.com/codeready-toolchain/member-operator/pull/400
- api: https://github.com/codeready-toolchain/api/pull/333
- sandbox-sre:https://github.com/codeready-toolchain/sandbox-sre/pull/874

### CODE REVIEW

File: scripts/add-cluster.sh [hunk 1]
```diff
@@ -97,6 +97,7 @@ rules:
   - "memberstatuses"
   - "toolchainclusters"
   - "useraccounts"
+  - "spacerequests"
```
Comment (xcoulon) [inline scripts/add-cluster.sh:100]: I don't remember why we need this here? Isn't it already generated and available in the member-operator repository? 🤔

Comment (mfrancisc) [reply scripts/add-cluster.sh:100]: I see this is giving permissions to `e2e-service-account` to manage spacerequests. 

But cannot find actually where the e2e tests are using this service account. 🤷‍♂️ 

maybe @MatousJobanek knows better how this Role(binding) it's being used.

Comment (MatousJobanek) [reply scripts/add-cluster.sh:100]: These ServiceAccounts are used for the ToolchainCluster CRs - as a representation of either host or member cluster.

As you know - the main usage is during the runtime when a host cluster needs to talk to a member or vice versa. For this we create:
* SA
* necessary (Cluster)Role(Bindings) so host (or member) has enough permissions. 

For this particular use case, we define the permissions that are needed for the communication between the operators/clusters, but not more than needed. Reasons:
* Avoid any potential privilege escalation
* Keep control over what host/member can access/do


The second use case is in e2e tests. When we initialize the test suite, we do it from the host cluster (or using the information that is in toolchain-host-operator namespace) so we can make it dynamic for one or more member installations. This is done using TolchainCluster CRs as well (but using different types to distinguish between standard ToolchainCluster and this e2e special one).
See the creation:
https://github.com/codeready-toolchain/toolchain-e2e/blob/master/make/test.mk#L233-L239
and initialization:
https://github.com/codeready-toolchain/toolchain-e2e/blob/master/testsupport/init.go#L152-L171

As you can imagine, for e2e we need much more permissions, so we have this part https://github.com/codeready-toolchain/toolchain-cicd/blob/master/scripts/add-cluster.sh#L149-L178 which grants cluster-admin permissions to the SA. Reason is simple - we don't have to propagate e2e test permissions to production (Cluster)Roles

**Review by MatousJobanek — APPROVED** [review]

File: scripts/add-cluster.sh (hunk 1 above)
Comment (mfrancisc) [reply scripts/add-cluster.sh:100]: Thanks for additional details, that's why we have separate `toolchaincluster` resources during `e2e`.

But I'm still struggling to understand where the `e2e-service-account` is being referenced/used :( 
I'm sure there is something I'm missing. Anyway we can take this discussion offline maybe!

This is my understanding about the e2e use case:

1. a token is being created: [https://github.com/codeready-toolchain/toolchain-cicd/blob/5fdcf455e10f292138416f5180cdb48c07b2b069/scripts/add-cluster.sh#L299](https://github.com/codeready-toolchain/toolchain-cicd/blob/5fdcf455e10f292138416f5180cdb48c07b2b069/scripts/add-cluster.sh#L299)
2. the openshift certificate is being retrieved: https://github.com/codeready-toolchain/toolchain-cicd/blob/5fdcf455e10f292138416f5180cdb48c07b2b069/scripts/add-cluster.sh#L311
3. with the token and the certificate a secret is being created: https://github.com/codeready-toolchain/toolchain-cicd/blob/5fdcf455e10f292138416f5180cdb48c07b2b069/scripts/add-cluster.sh#L344
4. the secret ref name is being added to the toolchaincluster spec: https://github.com/codeready-toolchain/toolchain-cicd/blob/5fdcf455e10f292138416f5180cdb48c07b2b069/scripts/add-cluster.sh#L367
5. the secret data is used to create a `restConfig`: https://github.com/codeready-toolchain/toolchain-common/blob/3415e8af365431e4fa5c24450d914a54de1e45c9/pkg/cluster/service.go#L172

**Pull request merged**